python311Packages.imageio: 2.32.0 -> 2.33.0

GaetanLepage opened the pull request:
## Description of changes

Changelog: https://github.com/imageio/imageio/releases/tag/v2.33.0

cc @Luflosi 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

GaetanLepage committed:
python311Packages.imageio: 2.32.0 -> 2.33.0

Changelog: https://github.com/imageio/imageio/releases/tag/v2.33.0

Luflosi commented:
Could you please fix the link to the changelog?

GaetanLepage commented:
> Could you please fix the link to the changelog?

Done !

Luflosi commented on pkgs/development/python-modules/imageio/default.nix:
```diff
@@ -32,24 +31,19 @@
-    inherit pname version;
-    hash = "sha256-5CWtNsYFMI2eptk+2nsJh5JgWbi4YiDhQqWZp5dRKN0=";
+  src = fetchFromGitHub {
+    owner = "imageio";
+    repo = "imageio";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-WoCycrJxo0vyV9LiWnEag1wbld3EJWu8mks8TnYt2+A=";
   };
```
Was there any particular reason you switched to GitHub? To be clear, I think this is a perfectly reasonable change, I'm just wondering.

NickCao replied:
> Was there any particular reason you switched to GitHub? To be clear, I think this is a perfectly reasonable change, I'm just wondering.

pypi tarballs are normally preprocessed, so not source codes in their natural habitat.

Luflosi reviewed (approved):
Please fix the version number in the commit and in the PR title, then this LGTM.

GaetanLepage commented:
> Please fix the version number in the commit and in the PR title, then this LGTM.

Done

mweinelt commented:
Result of `nixpkgs-review pr 268651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.stytra</li>
    <li>python311Packages.stytra.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>158 packages built:</summary>
  <ul>
    <li>deface</li>
    <li>deface.dist</li>
    <li>easyocr (python311Packages.easyocr)</li>
    <li>easyocr.dist (python311Packages.easyocr.dist)</li>
    <li>home-assistant-component-tests.dremel_3d_printer</li>
    <li>home-assistant-component-tests.environment_canada</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>intensity-normalization (python311Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python311Packages.intensity-normalization.dist)</li>
    <li>napari (python311Packages.napari)</li>
    <li>napari.dist (python311Packages.napari.dist)</li>
    <li>ome_zarr (python311Packages.ome-zarr)</li>
    <li>ome_zarr.dist (python311Packages.ome-zarr.dist)</li>
    <li>python310Packages.albumentations</li>
    <li>python310Packages.albumentations.dist</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.aplpy.dist</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.batchgenerators.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.dask-image.dist</li>
    <li>python310Packages.dremel3dpy</li>
    <li>python310Packages.dremel3dpy.dist</li>
    <li>python310Packages.easyocr</li>
    <li>python310Packages.easyocr.dist</li>
    <li>python310Packages.env-canada</li>
    <li>python310Packages.env-canada.dist</li>
    <li>python310Packages.glueviz</li>
    <li>python310Packages.glueviz.dist</li>
    <li>python310Packages.glymur</li>
    <li>python310Packages.glymur.dist</li>
    <li>python310Packages.gradio</li>
    <li>python310Packages.gradio.dist</li>
    <li>python310Packages.gymnasium</li>
    <li>python310Packages.gymnasium.dist</li>
    <li>python310Packages.handout</li>
    <li>python310Packages.handout.dist</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imagecorruptions.dist</li>
    <li>python310Packages.imageio</li>
    <li>python310Packages.imageio.dist</li>
    <li>python310Packages.insightface</li>
    <li>python310Packages.insightface.dist</li>
    <li>python310Packages.intensity-normalization</li>
    <li>python310Packages.intensity-normalization.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.lime</li>
    <li>python310Packages.lime.dist</li>
    <li>python310Packages.moviepy</li>
    <li>python310Packages.moviepy.dist</li>
    <li>python310Packages.mung</li>
    <li>python310Packages.mung.dist</li>
    <li>python310Packages.muscima</li>
    <li>python310Packages.muscima.dist</li>
    <li>python310Packages.napari</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-console.dist</li>
    <li>python310Packages.napari-svg</li>
    <li>python310Packages.napari-svg.dist</li>
    <li>python310Packages.napari.dist</li>
    <li>python310Packages.ome-zarr</li>
    <li>python310Packages.ome-zarr.dist</li>
    <li>python310Packages.omrdatasettools</li>
    <li>python310Packages.omrdatasettools.dist</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.pims.dist</li>
    <li>python310Packages.psd-tools</li>
    <li>python310Packages.psd-tools.dist</li>
    <li>python310Packages.pvextractor</li>
    <li>python310Packages.pvextractor.dist</li>
    <li>python310Packages.pyrender</li>
    <li>python310Packages.pyrender.dist</li>
    <li>python310Packages.pyvista</li>
    <li>python310Packages.pyvista.dist</li>
    <li>python310Packages.scikit-image</li>
    <li>python310Packages.scikit-image.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.slicedimage.dist</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.spectral-cube.dist</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.test-tube.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>python311Packages.albumentations</li>
    <li>python311Packages.albumentations.dist</li>
    <li>python311Packages.aplpy</li>
    <li>python311Packages.aplpy.dist</li>
    <li>python311Packages.batchgenerators</li>
    <li>python311Packages.batchgenerators.dist</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
    <li>python311Packages.dask-image</li>
    <li>python311Packages.dask-image.dist</li>
    <li>python311Packages.dremel3dpy</li>
    <li>python311Packages.dremel3dpy.dist</li>
    <li>python311Packages.env-canada</li>
    <li>python311Packages.env-canada.dist</li>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glueviz.dist</li>
    <li>python311Packages.glymur</li>
    <li>python311Packages.glymur.dist</li>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio.dist</li>
    <li>python311Packages.handout</li>
    <li>python311Packages.handout.dist</li>
    <li>python311Packages.imagecorruptions</li>
    <li>python311Packages.imagecorruptions.dist</li>
    <li>python311Packages.imageio</li>
    <li>python311Packages.imageio.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.lime.dist</li>
    <li>python311Packages.moviepy</li>
    <li>python311Packages.moviepy.dist</li>
    <li>python311Packages.mung</li>
    <li>python311Packages.mung.dist</li>
    <li>python311Packages.muscima</li>
    <li>python311Packages.muscima.dist</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-console.dist</li>
    <li>python311Packages.napari-svg</li>
    <li>python311Packages.napari-svg.dist</li>
    <li>python311Packages.omrdatasettools</li>
    <li>python311Packages.omrdatasettools.dist</li>
    <li>python311Packages.pims</li>
    <li>python311Packages.pims.dist</li>
    <li>python311Packages.psd-tools</li>
    <li>python311Packages.psd-tools.dist</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pvextractor.dist</li>
    <li>python311Packages.pyrender</li>
    <li>python311Packages.pyrender.dist</li>
    <li>python311Packages.pyvista</li>
    <li>python311Packages.pyvista.dist</li>
    <li>python311Packages.scikit-image</li>
    <li>python311Packages.scikit-image.dist</li>
    <li>python311Packages.slicedimage</li>
    <li>python311Packages.slicedimage.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
    <li>python311Packages.test-tube</li>
    <li>python311Packages.test-tube.dist</li>
    <li>tartube</li>
    <li>tartube-yt-dlp</li>
    <li>tartube-yt-dlp.dist</li>
    <li>tartube.dist</li>
  </ul>
</details>

mweinelt commented:
Can you check on the failing builds? I've already confirmed tts is still working. I just don't want to leak breakages onto master right now.

GaetanLepage commented:
> Can you check on the failing builds? I've already confirmed tts is still working. I just don't want to leak breakages onto master right now.

I have re-run a `nixpkgs-review` and here are the results:

Result of `nixpkgs-review pr 268651` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.stytra</li>
    <li>python311Packages.stytra.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
  </ul>
</details>
<details>
  <summary>168 packages built:</summary>
  <ul>
    <li>deface</li>
    <li>deface.dist</li>
    <li>easyocr (python311Packages.easyocr)</li>
    <li>easyocr.dist (python311Packages.easyocr.dist)</li>
    <li>home-assistant-component-tests.dremel_3d_printer</li>
    <li>home-assistant-component-tests.environment_canada</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>intensity-normalization (python311Packages.intensity-normalization)</li>
    <li>intensity-normalization.dist (python311Packages.intensity-normalization.dist)</li>
    <li>napari (python311Packages.napari)</li>
    <li>napari.dist (python311Packages.napari.dist)</li>
    <li>ome_zarr (python311Packages.ome-zarr)</li>
    <li>ome_zarr.dist (python311Packages.ome-zarr.dist)</li>
    <li>python310Packages.albumentations</li>
    <li>python310Packages.albumentations.dist</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.aplpy.dist</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.batchgenerators.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.dask-image.dist</li>
    <li>python310Packages.dremel3dpy</li>
    <li>python310Packages.dremel3dpy.dist</li>
    <li>python310Packages.easyocr</li>
    <li>python310Packages.easyocr.dist</li>
    <li>python310Packages.env-canada</li>
    <li>python310Packages.env-canada.dist</li>
    <li>python310Packages.glueviz</li>
    <li>python310Packages.glueviz.dist</li>
    <li>python310Packages.glymur</li>
    <li>python310Packages.glymur.dist</li>
    <li>python310Packages.gradio</li>
    <li>python310Packages.gradio.dist</li>
    <li>python310Packages.gymnasium</li>
    <li>python310Packages.gymnasium.dist</li>
    <li>python310Packages.handout</li>
    <li>python310Packages.handout.dist</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imagecorruptions.dist</li>
    <li>python310Packages.imageio</li>
    <li>python310Packages.imageio.dist</li>
    <li>python310Packages.insightface</li>
    <li>python310Packages.insightface.dist</li>
    <li>python310Packages.intensity-normalization</li>
    <li>python310Packages.intensity-normalization.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.lime</li>
    <li>python310Packages.lime.dist</li>
    <li>python310Packages.moviepy</li>
    <li>python310Packages.moviepy.dist</li>
    <li>python310Packages.mung</li>
    <li>python310Packages.mung.dist</li>
    <li>python310Packages.muscima</li>
    <li>python310Packages.muscima.dist</li>
    <li>python310Packages.napari</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-console.dist</li>
    <li>python310Packages.napari-svg</li>
    <li>python310Packages.napari-svg.dist</li>
    <li>python310Packages.napari.dist</li>
    <li>python310Packages.ome-zarr</li>
    <li>python310Packages.ome-zarr.dist</li>
    <li>python310Packages.omrdatasettools</li>
    <li>python310Packages.omrdatasettools.dist</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.pims.dist</li>
    <li>python310Packages.psd-tools</li>
    <li>python310Packages.psd-tools.dist</li>
    <li>python310Packages.pvextractor</li>
    <li>python310Packages.pvextractor.dist</li>
    <li>python310Packages.pyrender</li>
    <li>python310Packages.pyrender.dist</li>
    <li>python310Packages.pyvista</li>
    <li>python310Packages.pyvista.dist</li>
    <li>python310Packages.scikit-image</li>
    <li>python310Packages.scikit-image.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.slicedimage.dist</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.spectral-cube.dist</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.test-tube.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>python311Packages.albumentations</li>
    <li>python311Packages.albumentations.dist</li>
    <li>python311Packages.aplpy</li>
    <li>python311Packages.aplpy.dist</li>
    <li>python311Packages.batchgenerators</li>
    <li>python311Packages.batchgenerators.dist</li>
    <li>python311Packages.caffe</li>
    <li>python311Packages.caffe.bin</li>
    <li>python311Packages.dask-image</li>
    <li>python311Packages.dask-image.dist</li>
    <li>python311Packages.dremel3dpy</li>
    <li>python311Packages.dremel3dpy.dist</li>
    <li>python311Packages.env-canada</li>
    <li>python311Packages.env-canada.dist</li>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glueviz.dist</li>
    <li>python311Packages.glymur</li>
    <li>python311Packages.glymur.dist</li>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.handout</li>
    <li>python311Packages.handout.dist</li>
    <li>python311Packages.imagecorruptions</li>
    <li>python311Packages.imagecorruptions.dist</li>
    <li>python311Packages.imageio</li>
    <li>python311Packages.imageio.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.lime.dist</li>
    <li>python311Packages.moviepy</li>
    <li>python311Packages.moviepy.dist</li>
    <li>python311Packages.mung</li>
    <li>python311Packages.mung.dist</li>
    <li>python311Packages.muscima</li>
    <li>python311Packages.muscima.dist</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-console.dist</li>
    <li>python311Packages.napari-svg</li>
    <li>python311Packages.napari-svg.dist</li>
    <li>python311Packages.omrdatasettools</li>
    <li>python311Packages.omrdatasettools.dist</li>
    <li>python311Packages.pims</li>
    <li>python311Packages.pims.dist</li>
    <li>python311Packages.psd-tools</li>
    <li>python311Packages.psd-tools.dist</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pvextractor.dist</li>
    <li>python311Packages.pyrender</li>
    <li>python311Packages.pyrender.dist</li>
    <li>python311Packages.pyvista</li>
    <li>python311Packages.pyvista.dist</li>
    <li>python311Packages.scikit-image</li>
    <li>python311Packages.scikit-image.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.slicedimage</li>
    <li>python311Packages.slicedimage.dist</li>
    <li>python311Packages.spectral-cube</li>
    <li>python311Packages.spectral-cube.dist</li>
    <li>python311Packages.test-tube</li>
    <li>python311Packages.test-tube.dist</li>
    <li>tartube</li>
    <li>tartube-yt-dlp</li>
    <li>tartube-yt-dlp.dist</li>
    <li>tartube.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

GaetanLepage commented:
`bsuite` is failing to build on `master` as well because of its dependency `plotnine` which is broken.

mweinelt commented:
Let's wait for ofborg and then go for the merge.

GaetanLepage commented:
It fails to build on darwin, but this already the case on `master`.

mweinelt commented:
Can you find out which test crashes and disable it on darwin?

GaetanLepage commented:
> Can you find out which test crashes and disable it on darwin?

What is broken en darwin is not `imageio` itself but `python311Packages.av`.
I will try to fix it.

mweinelt commented:
> tests/test_pyav.py ...Fatal Python error: Segmentation fault

That's the pyav integration tests in imageio. But pyav seems to also crash on nixpkgs/trunk on hydra, but it doesn't here.

GaetanLepage commented:
Ok, I must admit that I have no clue on how to fix that...
```
tests/test_deprecation.py ..                                             [ 32%]
tests/test_dictionary.py .                                               [ 32%]
tests/test_encode.py ..Fatal Python error: Segmentation fault

Current thread 0x00000001e2a46080 (most recent call first):
  File "/private/tmp/nix-build-python3.11-av-10.0.0.drv-0/source/tests/test_encode.py", line 164 in test_encoding_with_pts
  File "/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/lib/python3.11/unittest/case.py", line 579 in _callTestMethod
  File "/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/lib/python3.11/unittest/case.py", line 623 in run
  File "/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/lib/python3.11/unittest/case.py", line 678 in __call__
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/unittest.py", line 333 in runtest
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 169 in pytest_runtest_call
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 262 in <lambda>
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 341 in from_call
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 261 in call_runtest_hook
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 222 in call_and_report
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 133 in runtestprotocol
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/main.py", line 350 in pytest_runtestloop
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/main.py", line 325 in _main
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/main.py", line 271 in wrap_session
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/main.py", line 318 in pytest_cmdline_main
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/nix/store/7jjjp2m5v063cpccn2imn3kx6k8y74nx-python3.11-pluggy-1.3.0/lib/python3.11/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/config/__init__.py", line 169 in main
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/_pytest/config/__init__.py", line 192 in console_main
  File "/nix/store/y6xaw9jxm878fb2x68z1j0m0p9qkx40k-python3.11-pytest-7.4.3/lib/python3.11/site-packages/pytest/__main__.py", line 5 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main

Extension modules: av._core, av.logging, av.bytesource, av.buffer, av.audio.format, av.enum, av.error, av.utils, av.option, av.descriptor, av.container.pyio, av.dictionary, av.format, av.stream, av.container.streams, av.sidedata.motionvectors, av.sidedata.sidedata, av.packet, av.container.input, av.container.output, av.container.core, av.codec.context, av.video.format, av.video.reformatter, av.plane, av.video.plane, av.video.frame, av.video.stream, av.codec.codec, av.frame, av.audio.layout, av.audio.plane, av.audio.frame, av.audio.stream, av.audio.fifo, av.filter.pad, av.filter.link, av.filter.context, av.filter.graph, av.filter.filter, av.audio.resampler, PIL._imaging, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, av.subtitles.subtitle, av.video.codeccontext, av.audio.codeccontext (total: 58)
/nix/store/2pqqdn2grr44ch4337jaqrwlzc7myv8m-pytest-check-hook/nix-support/setup-hook: line 53: 62860 Segmentation fault: 11  /nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 -m pytest -k "not test_writing_to_custom_io" --ignore="tests/test_doctests.py" --ignore="tests/test_timeout.py" --deselect=tests/test_audiofifo.py::TestAudioFifo::test_data --deselect=tests/test_codec_context.py::TestCodecContext::test_codec_tag --deselect=tests/test_codec_context.py::TestCodecContext::test_parse --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_aac --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_dnxhd --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_dvvideo --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_h264 --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_mjpeg --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_mp2 --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_mpeg1video --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_mpeg4 --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_pcm_s24le --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_png --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_tiff --deselect=tests/test_codec_context.py::TestEncoding::test_encoding_xvid --deselect=tests/test_decode.py::TestDecode::test_decode_audio_sample_count --deselect=tests/test_decode.py::TestDecode::test_decoded_motion_vectors --deselect=tests/test_decode.py::TestDecode::test_decoded_motion_vectors_no_flag --deselect=tests/test_decode.py::TestDecode::test_decoded_time_base --deselect=tests/test_decode.py::TestDecode::test_decoded_video_frame_count --deselect=tests/test_encode.py::TestBasicAudioEncoding::test_transcode --deselect=tests/test_file_probing.py::TestAudioProbe::test_container_probing --deselect=tests/test_file_probing.py::TestAudioProbe::test_stream_probing --deselect=tests/test_file_probing.py::TestDataProbe::test_container_probing --deselect=tests/test_file_probing.py::TestDataProbe::test_stream_probing --deselect=tests/test_file_probing.py::TestSubtitleProbe::test_container_probing --deselect=tests/test_file_probing.py::TestSubtitleProbe::test_stream_probing --deselect=tests/test_file_probing.py::TestVideoProbe::test_container_probing --deselect=tests/test_file_probing.py::TestVideoProbe::test_stream_probing --deselect=tests/test_python_io.py::TestPythonIO::test_reading_from_buffer --deselect=tests/test_python_io.py::TestPythonIO::test_reading_from_buffer_no_see --deselect=tests/test_python_io.py::TestPythonIO::test_reading_from_file --deselect=tests/test_python_io.py::TestPythonIO::test_reading_from_pipe_readonly --deselect=tests/test_python_io.py::TestPythonIO::test_reading_from_write_readonl --deselect=tests/test_seek.py::TestSeek::test_decode_half --deselect=tests/test_seek.py::TestSeek::test_seek_end --deselect=tests/test_seek.py::TestSeek::test_seek_float --deselect=tests/test_seek.py::TestSeek::test_seek_int64 --deselect=tests/test_seek.py::TestSeek::test_seek_middle --deselect=tests/test_seek.py::TestSeek::test_seek_start --deselect=tests/test_seek.py::TestSeek::test_stream_seek --deselect=tests/test_streams.py::TestStreams::test_selection --deselect=tests/test_streams.py::TestStreams::test_stream_tuples --deselect=tests/test_subtitles.py::TestSubtitle::test_movtext --deselect=tests/test_subtitles.py::TestSubtitle::test_vobsub --deselect=tests/test_videoframe.py::TestVideoFrameImage::test_roundtrip
/nix/store/2a1pjx5crgygz7wbvl1lmp49n6apfp8r-stdenv-darwin/setup: line 1568: pop_var_context: head of shell_variables not a function context
```

mweinelt committed:
python311Packages.av: disable crashing tests on darwin

mweinelt committed:
python311Packages.imageio: disable failing tests on darwin

One test is crashing the interpreter, same as on pyav.

The others require ffmpeg with h264 encoding support, which for some
reason is not available on darwin.

mweinelt commented:
The most straight forward way to find out what test is failing is to add verbosity to `pytestFlagsArray`.

mweinelt commented:
@ofborg eval
